ocaml/alerts/certificate/certificate_check.ml

```diff
@@ -61,10 +59,23 @@ let execute rpc session (host, alert) =
 
 let alert rpc session =
   let now = Unix.time () in
+  (* Message starting with [host_server_certificate_\{expiring,expired\}] may
+     need to be refreshed, gather them just once *)
+  let previous_messages = XenAPI.Message.get_all_records rpc session
+  |> List.filter (fun (ref, record) ->
+      let expiring_or_expired name =
```

Comment:
I think `message` would be a better name but I understand it comes from the name of the field.